### ISSUE TITLE
[BUG] - Daedalus 3.3.2  cardano-node crashed (Ubuntu 20.04)

### ISSUE THREAD

**Issue by rekuenkdr** [input-output-hk/cardano-node#2413]
**Internal/External**
*External* 

**Area**
*Other*  :  Daedalus wallet.

**Summary**
I have downloaded the latest daedalus-3.3.1-mainnet-16693.bin and it fails to connect to cardano-node on first start under Ubuntu 20.04.2 LTS.

**Update: I have downloaded Daedalus 3.3.2#16919 and the issue still remains the same.**

**Steps to reproduce**

1. Download '[Daedalus 3.3.1](https://update-cardano-mainnet.iohk.io/daedalus-3.3.1-mainnet-16693.bin?t=1614261561177)' or Daedalus 3.3.2
2. Install the wallet 
3. Start the wallet for the first time
4. Daedalus starts, the screen stars to flicker and is totally unusable, it says cardano node crashed.
5. The developer console also has this javascript error:

```
[2021-02-25T14:06:16.696Z] [daedalus.*mainnet*:error:ipcRenderer] {
  "msg": "AdaApi::getNetworkInfo error",
  "data": {
    "error": {}
  }
}
(anonymous) | @ | renderer.js:34
```

**Expected behavior**
The wallet should start and connect to the node, to start syncing.

**System info:**
- OS: Ubuntu
- Version  20.04.2 LTS
- Daedalus version 3.3.1 Mainnet

**Screenshots and attachments**
![Captura de pantalla de 2021-02-25 15-06-41](https://user-images.githubusercontent.com/42606743/109165889-5fb0bb80-777c-11eb-95db-c5cb199ee9c5.png)
![Captura de pantalla de 2021-02-25 15-08-34](https://user-images.githubusercontent.com/42606743/109165893-60495200-777c-11eb-81ad-18178620c040.png)

**Additional context**
I provide the logs of the application.
[logs-2021-02-25T120733.0742Z.zip](https://github.com/input-output-hk/cardano-node/files/6043535/logs-2021-02-25T120733.0742Z.zip)

**Comment by K3rnelPani1c** [comment]
same issue

**Comment by adaasch** [comment]
Me too

**Comment by rekuenkdr** [comment]
I have updated to Daedalus 3.3.2 and the issue still remains the same.

**Comment by Sheckay** [comment]
I went through everything I could to get it installed on to Linux Mint. Did not work. But I heard that it's easy with Ubuntu. So I went through that install. And the install works but it doesn't run. I get the message "Cardano node crashed" every 5 seconds or so. The screen refreshes each time.

**Comment by AJG85** [comment]
TL;DR: Could an admin or maintainer move this thread to https://github.com/input-output-hk/daedalus issues so it gets some attention?

Daedalus is unusable on Ubuntu 20.04.2 LTS Linux in it's current state. You can't even agree to the terms before it crashes. That said this certainly is not the right place for this bug report as this repository is for the cardano node itself!

In addition to the getNetworkInfo() error on the embedded browser console there is no activity on the network tab. Apparently it is a JavaScript application written with Electron thus the browser in the "desktop app"

When launching from the terminal you can see an additional errors claiming the updater file is missing. This may be an incomplete feature for the future.

I also noticed the canberra-gtk-module was missing from my system. This can be easily fixed by running `sudo apt install libcanberra-gtk-module` however it won't address the instability of Daedalus trying to connect the node.

There are multiple warnings about using deprecated and insecure buffers which seems to indicate an older or outdated version of the node is packaged in the bzip perhaps?
`(node:15124) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.`

The node.log is empty in my case (as it never actually connects) but the most interesting error in the cardano-wallet.log seemed to be right after loading the SQL lite database for stake pools:
`ntp client experienced error Network.Socket.ByteString.sendManyTo: does not exist (Network is unreachable) when sending packet` which I thought may indicate a clock or local sync issue. I installed the ntp client and ensured it was running with `ntpq -p` but this had no impact either.

At this point I've pretty much run out of ideas other than using Windoze in Virtual Box or cloning https://github.com/input-output-hk/cardano-wallet and making my own UI perhaps ...

**Comment by rekuenkdr** [comment]
> TL;DR: Could an admin or maintainer move this thread to https://github.com/input-output-hk/daedalus issues so it gets some attention?
> 
> Daedalus is unusable on Ubuntu 20.04.2 LTS Linux in it's current state. You can't even agree to the terms before it crashes. That said this certainly is not the right place for this bug report as this repository is for the cardano node itself!
> 
> In addition to the getNetworkInfo() error on the embedded browser console there is no activity on the network tab. Apparently it is a JavaScript application written with Electron thus the browser in the "desktop app"
> 
> When launching from the terminal you can see an additional errors claiming the updater file is missing. This may be an incomplete feature for the future.
> 
> I also noticed the canberra-gtk-module was missing from my system. This can be easily fixed by running `sudo apt install libcanberra-gtk-module` however it won't address the instability of Daedalus trying to connect the node.
> 
> There are multiple warnings about using deprecated and insecure buffers which seems to indicate an older or outdated version of the node is packaged in the bzip perhaps?
> `(node:15124) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.`
> 
> The node.log is empty in my case (as it never actually connects) but the most interesting error in the cardano-wallet.log seemed to be right after loading the SQL lite database for stake pools:
> `ntp client experienced error Network.Socket.ByteString.sendManyTo: does not exist (Network is unreachable) when sending packet` which I thought may indicate a clock or local sync issue. I installed the ntp client and ensured it was running with `ntpq -p` but this had no impact either.
> 
> At this point I've pretty much run out of ideas other than using Windoze in Virtual Box or cloning https://github.com/input-output-hk/cardano-wallet and making my own UI perhaps ...

I reported this on the daedalus repository first, but they ask me to post it here,  because it was a node problem according to them.

**Comment by stany9g** [comment]
same

**Comment by AJG85** [comment]
> > TL;DR: Could an admin or maintainer move this thread to https://github.com/input-output-hk/daedalus issues so it gets some attention?
> > Daedalus is unusable on Ubuntu 20.04.2 LTS Linux in it's current state. You can't even agree to the terms before it crashes. That said this certainly is not the right place for this bug report as this repository is for the cardano node itself!
> > In addition to the getNetworkInfo() error on the embedded browser console there is no activity on the network tab. Apparently it is a JavaScript application written with Electron thus the browser in the "desktop app"
> > When launching from the terminal you can see an additional errors claiming the updater file is missing. This may be an incomplete feature for the future.
> > I also noticed the canberra-gtk-module was missing from my system. This can be easily fixed by running `sudo apt install libcanberra-gtk-module` however it won't address the instability of Daedalus trying to connect the node.
> > There are multiple warnings about using deprecated and insecure buffers which seems to indicate an older or outdated version of the node is packaged in the bzip perhaps?
> > `(node:15124) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.`
> > The node.log is empty in my case (as it never actually connects) but the most interesting error in the cardano-wallet.log seemed to be right after loading the SQL lite database for stake pools:
> > `ntp client experienced error Network.Socket.ByteString.sendManyTo: does not exist (Network is unreachable) when sending packet` which I thought may indicate a clock or local sync issue. I installed the ntp client and ensured it was running with `ntpq -p` but this had no impact either.
> > At this point I've pretty much run out of ideas other than using Windoze in Virtual Box or cloning https://github.com/input-output-hk/cardano-wallet and making my own UI perhaps ...
> 
> I reported this on the daedalus repository first, but they asked me to post it here, because it was a node problem according to them.
> 
> [input-output-hk/daedalus#2403](https://github.com/input-output-hk/daedalus/issues/2403)

Interesting, since the node.log is empty and it never actually connects I would think it's something wrong with the way the node is packaged within Daedalus for Linux or it's an environment issue with Ubuntu maybe. If the problem was with the node itself I would expect other wallets to have issues as well. FYI I can confirm same version of Daedalus works fine in a Windows 10 VM ;-)

**Comment by AJG85** [comment]
Update: Daedalus Mainnet 4.0.4 Linux build 17475 resolves this issue.

Kudos!

**Comment by rekuenkdr** [comment]
I can confirm that Daedalus Mainnet 4.0.4 Linux [build 17475](https://update-cardano-mainnet.iohk.io/daedalus-4.0.4-mainnet-17475.bin?t=1617530603169) solves the issue for me as well.

So if anyone is having the same issue please upgrade your wallet to this version.
I'm closing the issue, if anyone is still having trouble after updating, leave a comment and I will reopen it.
